contracts/utils/contracts/src/LibFractions.sol

```diff
@@ -27,12 +30,25 @@ library LibFractions {
             uint256 denominator
         )
     {
+        if (n1 == 0) {
+            return (numerator = n2, denominator = d2);
+        }
+        if (n2 == 0) {
+            return (numerator = n1, denominator = d1);
+        }
         numerator = n1
             .safeMul(d2)
             .safeAdd(n2.safeMul(d1));
-
         denominator = d1.safeMul(d2);
-        return (numerator, denominator);
+
+        // If either the numerator or the denominator are > RESCALE_THRESHOLD,
```

Comment:
Hmm. Could also move it out to a separate `rescaleFractions(threshold)` function. We should probably also drop all the `*Fractions` suffixes to these functions since it's stuttering.